From e7a105cd60e1b17ae5cc20a7a017db9f37cd1d92 Mon Sep 17 00:00:00 2001
From: ChenXiaoSong <chenxiaosong@kylinos.cn>
Date: Mon, 25 May 2026 07:09:33 +0000
Subject: [PATCH 4/5] fanotify: export fsnotify_add_mark_locked

Signed-off-by: ChenXiaoSong <chenxiaosong@kylinos.cn>
---
 fs/notify/mark.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/fs/notify/mark.c b/fs/notify/mark.c
index e256b420100d..a7af585d3721 100644
--- a/fs/notify/mark.c
+++ b/fs/notify/mark.c
@@ -982,6 +982,7 @@ int fsnotify_add_mark_locked(struct fsnotify_mark *mark,
 	fsnotify_put_mark(mark);
 	return ret;
 }
+EXPORT_SYMBOL_GPL(fsnotify_add_mark_locked);
 
 int fsnotify_add_mark(struct fsnotify_mark *mark, void *obj,
 		      unsigned int obj_type, int add_flags)
-- 
2.54.0

