From 7d260f90cbc7f690c1ff37ca23009a7cec642f34 Mon Sep 17 00:00:00 2001
From: ChenXiaoSong <chenxiaosong@kylinos.cn>
Date: Tue, 16 Jun 2026 12:03:03 +0000
Subject: [PATCH 09/23] smb/server: ksmbd_notify_filter_to_mask --> inotify_map

Signed-off-by: ChenXiaoSong <chenxiaosong@kylinos.cn>
---
 fs/smb/server/notify.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/fs/smb/server/notify.c b/fs/smb/server/notify.c
index e1b079441e05..37db23772a1a 100644
--- a/fs/smb/server/notify.c
+++ b/fs/smb/server/notify.c
@@ -59,6 +59,19 @@ static const struct {
 	{ FILE_NOTIFY_CHANGE_SECURITY, FS_ATTRIB },
 };
 
+static u32 ksmbd_notify_filter_to_mask(u32 filter)
+{
+	size_t i;
+	u32 mask = FS_EVENT_ON_CHILD;
+
+	for (i = 0; i < ARRAY_SIZE(ksmbd_notify_mapping); i++) {
+		if (ksmbd_notify_mapping[i].notify_mask & filter)
+			mask |= ksmbd_notify_mapping[i].fsnotify_mask;
+	}
+
+	return mask;
+}
+
 static void smb2_cancel_notify(void **argv)
 {
 	struct ksmbd_notify_ctx *ctx = argv[0];
-- 
2.54.0

