From 9c4f9efd48be8ba8e49c2fdef8f97f413844270f Mon Sep 17 00:00:00 2001
From: ChenXiaoSong <chenxiaosong@kylinos.cn>
Date: Thu, 1 May 2025 19:01:35 +0800
Subject: [PATCH 2/5] mydebug: mainline

Signed-off-by: ChenXiaoSong <chenxiaosong@kylinos.cn>
---
 Kbuild | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Kbuild b/Kbuild
index 13324b4bbe23..37609c486fcf 100644
--- a/Kbuild
+++ b/Kbuild
@@ -95,6 +95,7 @@ obj-y			+= kernel/
 obj-y			+= certs/
 obj-y			+= mm/
 obj-y			+= fs/
+obj-y			+= mydebug/
 obj-y			+= ipc/
 obj-y			+= security/
 obj-y			+= crypto/
-- 
2.43.0

