From 96cf1858969889eab3d29c1c9fd0b654eed74a55 Mon Sep 17 00:00:00 2001
From: ChenXiaoSong <chenxiaosong@kylinos.cn>
Date: Thu, 8 Jan 2026 09:43:38 +0800
Subject: [PATCH 4/5] smb/client: cleanups in SMB2_change_notify()

Signed-off-by: ChenXiaoSong <chenxiaosong@kylinos.cn>
---
 fs/smb/client/smb2pdu.c | 48 +++++++++++++++++++++--------------------
 1 file changed, 25 insertions(+), 23 deletions(-)

diff --git a/fs/smb/client/smb2pdu.c b/fs/smb/client/smb2pdu.c
index 5d57c895ca37..be98d8b6258f 100644
--- a/fs/smb/client/smb2pdu.c
+++ b/fs/smb/client/smb2pdu.c
@@ -4110,31 +4110,33 @@ SMB2_change_notify(const unsigned int xid, struct cifs_tcon *tcon,
 		cifs_stats_fail_inc(tcon, SMB2_CHANGE_NOTIFY_HE);
 		trace_smb3_notify_err(xid, persistent_fid, tcon->tid, ses->Suid,
 				(u8)watch_tree, completion_filter, rc);
-	} else {
-		trace_smb3_notify_done(xid, persistent_fid, tcon->tid,
-			ses->Suid, (u8)watch_tree, completion_filter);
-		/* validate that notify information is plausible */
-		if ((rsp_iov.iov_base == NULL) ||
-		    (rsp_iov.iov_len < sizeof(struct smb2_change_notify_rsp) + 1))
-			goto cnotify_exit;
-
-		smb_rsp = (struct smb2_change_notify_rsp *)rsp_iov.iov_base;
-
-		rc = smb2_validate_iov(le16_to_cpu(smb_rsp->OutputBufferOffset),
-				le32_to_cpu(smb_rsp->OutputBufferLength),
-				&rsp_iov,
-				sizeof(struct file_notify_information));
-		if (rc)
-			goto cnotify_exit;
+		goto cnotify_exit;
+	}
 
-		*out_data = kmemdup((char *)smb_rsp + le16_to_cpu(smb_rsp->OutputBufferOffset),
-				le32_to_cpu(smb_rsp->OutputBufferLength), GFP_KERNEL);
-		if (*out_data == NULL) {
-			rc = -ENOMEM;
-			goto cnotify_exit;
-		} else if (plen)
-			*plen = le32_to_cpu(smb_rsp->OutputBufferLength);
+	trace_smb3_notify_done(xid, persistent_fid, tcon->tid,
+		ses->Suid, (u8)watch_tree, completion_filter);
+	/* validate that notify information is plausible */
+	if ((rsp_iov.iov_base == NULL) ||
+	    (rsp_iov.iov_len < sizeof(struct smb2_change_notify_rsp) + 1))
+		goto cnotify_exit;
+
+	smb_rsp = (struct smb2_change_notify_rsp *)rsp_iov.iov_base;
+
+	rc = smb2_validate_iov(le16_to_cpu(smb_rsp->OutputBufferOffset),
+			le32_to_cpu(smb_rsp->OutputBufferLength),
+			&rsp_iov,
+			sizeof(struct file_notify_information));
+	if (rc)
+		goto cnotify_exit;
+
+	*out_data = kmemdup((char *)smb_rsp + le16_to_cpu(smb_rsp->OutputBufferOffset),
+			le32_to_cpu(smb_rsp->OutputBufferLength), GFP_KERNEL);
+	if (*out_data == NULL) {
+		rc = -ENOMEM;
+		goto cnotify_exit;
 	}
+	if (plen)
+		*plen = le32_to_cpu(smb_rsp->OutputBufferLength);
 
  cnotify_exit:
 	if (rqst.rq_iov)
-- 
2.43.0

