From cf86ba9746112b5faca9db36cd401de25b43d592 Mon Sep 17 00:00:00 2001
From: ChenXiaoSong <chenxiaosong@kylinos.cn>
Date: Thu, 28 May 2026 11:42:54 +0000
Subject: [PATCH 01/23] smb/server: enable notify debug

Signed-off-by: ChenXiaoSong <chenxiaosong@kylinos.cn>
---
 fs/smb/server/server.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/smb/server/server.c b/fs/smb/server/server.c
index 36feda7e0942..1f7a73e8d7e0 100644
--- a/fs/smb/server/server.c
+++ b/fs/smb/server/server.c
@@ -24,7 +24,7 @@
 #include "stats.h"
 #include "compress.h"
 
-int ksmbd_debug_types;
+int ksmbd_debug_types = KSMBD_DEBUG_NOTIFY;
 
 struct ksmbd_server_config server_conf;
 
-- 
2.54.0

