From 14e8e57999cd44ff921e2c3338c91b0755efb4cf Mon Sep 17 00:00:00 2001
From: ChenXiaoSong <chenxiaosong@kylinos.cn>
Date: Fri, 29 May 2026 00:13:08 +0000
Subject: [PATCH 05/23] smb/server: fix hung task

Signed-off-by: ChenXiaoSong <chenxiaosong@kylinos.cn>
---
 fs/smb/server/notify.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/smb/server/notify.c b/fs/smb/server/notify.c
index 6c0f2d428045..6ccc700201cb 100644
--- a/fs/smb/server/notify.c
+++ b/fs/smb/server/notify.c
@@ -65,7 +65,7 @@ int smb2_setup_notify(struct ksmbd_work *work,
 
 	smb2_send_interim_resp(work, STATUS_PENDING);
 
-	wait_event(ctx->waitq, work->state != KSMBD_WORK_ACTIVE);
+	wait_event_interruptible(ctx->waitq, work->state != KSMBD_WORK_ACTIVE);
 
 	if (work->state == KSMBD_WORK_CANCELLED) {
 		rsp->hdr.Status = STATUS_CANCELLED;
-- 
2.54.0

