From 2345295c974f645c67111ace5fa2fff3883da3bf Mon Sep 17 00:00:00 2001
From: ChenXiaoSong <chenxiaosong@kylinos.cn>
Date: Tue, 16 Jun 2026 11:58:53 +0000
Subject: [PATCH 08/23] smb/server: ksmbd_notify_mapping --> inotify_mapping

Signed-off-by: ChenXiaoSong <chenxiaosong@kylinos.cn>
---
 fs/smb/server/notify.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/fs/smb/server/notify.c b/fs/smb/server/notify.c
index 8396895d88f6..e1b079441e05 100644
--- a/fs/smb/server/notify.c
+++ b/fs/smb/server/notify.c
@@ -43,6 +43,22 @@ struct ksmbd_notify_ctx {
 	bool watching;
 };
 
+static const struct {
+	u32 notify_mask;
+	u32 fsnotify_mask;
+} ksmbd_notify_mapping[] = {
+	{ FILE_NOTIFY_CHANGE_FILE_NAME,
+	  FS_CREATE | FS_DELETE | FS_MOVED_FROM | FS_MOVED_TO },
+	{ FILE_NOTIFY_CHANGE_DIR_NAME,
+	  FS_CREATE | FS_DELETE | FS_MOVED_FROM | FS_MOVED_TO },
+	{ FILE_NOTIFY_CHANGE_ATTRIBUTES,
+	  FS_ATTRIB | FS_MOVED_FROM | FS_MOVED_TO | FS_MODIFY },
+	{ FILE_NOTIFY_CHANGE_LAST_WRITE, FS_ATTRIB },
+	{ FILE_NOTIFY_CHANGE_LAST_ACCESS, FS_ATTRIB },
+	{ FILE_NOTIFY_CHANGE_EA, FS_ATTRIB },
+	{ FILE_NOTIFY_CHANGE_SECURITY, FS_ATTRIB },
+};
+
 static void smb2_cancel_notify(void **argv)
 {
 	struct ksmbd_notify_ctx *ctx = argv[0];
-- 
2.54.0

