From 245b223c9608f9acdf3d848be2dada328742b41e Mon Sep 17 00:00:00 2001
From: ChenXiaoSong <chenxiaosong@kylinos.cn>
Date: Tue, 16 Jun 2026 12:27:54 +0000
Subject: [PATCH 13/23] smb/server: ksmbd_notify_unwatch --> watch_destructor

Signed-off-by: ChenXiaoSong <chenxiaosong@kylinos.cn>
---
 fs/smb/server/notify.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/fs/smb/server/notify.c b/fs/smb/server/notify.c
index c50bb4ec0864..bdd64fcc9c55 100644
--- a/fs/smb/server/notify.c
+++ b/fs/smb/server/notify.c
@@ -175,6 +175,25 @@ static int ksmbd_notify_watch(struct ksmbd_notify_ctx *ctx, struct inode *dir)
 	return rc;
 }
 
+static void ksmbd_notify_unwatch(struct ksmbd_notify_ctx *ctx)
+{
+	struct fsnotify_group *group = ctx->group;
+	struct ksmbd_notify_mark *imark = ctx->inode_mark;
+
+	ctx->watching = false;
+
+	if (!group)
+		return;
+
+	if (imark) {
+		fsnotify_destroy_mark(&imark->fsn_mark, group);
+		ctx->inode_mark = NULL;
+	}
+
+	fsnotify_put_group(group);
+	ctx->group = NULL;
+}
+
 static void smb2_cancel_notify(void **argv)
 {
 	struct ksmbd_notify_ctx *ctx = argv[0];
-- 
2.54.0

